frontend/src/pages/Player/PlayerHook/PlayerState.ts

```diff
@@ -346,6 +346,10 @@ export const PlayerReducer = (
 			break
 		case PlayerActionType.setTime:
 			s.time = action.time
+			s.currentUrl = findLatestUrl(
+				getAllUrlEvents(events),
+				action.time + s.sessionMetadata.startTime,
+			)
```

Comment:
a bit worried that doing this on each debounced rerender would be slow if there are a lot of events to search. doesn't seem from profiling like this is a bottleneck though, but something to keep in mind

Reply:
For sure! I think this method (`findLatestUrl`) should be called a little more than the current state (called in `PlayerActionType.updateCurrentUrl`). It will still get called on start and pause, but this is just adding to when the user manually changes the time.

Reply:
ah cool, thanks!